x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/store/reducer.ts

```diff
@@ -131,6 +134,18 @@ export const endpointListReducer: ImmutableReducer<EndpointState, AppAction> = (
       ...state,
       endpointsExist: action.payload,
     };
+  } else if (action.type === 'userUpdatedEndpointListRefreshOptions') {
```

Comment:
I was wondering this before when I looked at the definition of this action's payload. Why not make the type stronger instead of optional properties? Example:

```typescript
type payload = {
    // refresh interval optional if off
    isAutoRefreshEnabled: false,
    autoRefreshInterval?: number
} | 
{ // both required if auto refresh is on
    isAutoRefreshEnabled: true,
    autoRefreshInterval: number
}
```
With the above, this code here could be simplified to `...action.payload` right?

Another alternative to avoiding the conditional checks:

```typescript
isAutoRefreshEnabled: action.payload.isAutoRefreshEnabled ?? state.isAutoRefreshEnabled,
autoRefreshInternal: action.payload.autoRefreshInterval ?? state.autoRefreshInterval
```

Reply:
will make the change.  I wanted it to be possible to specify one or the other while preserving the last value if it's not specified.

I think I prefer this one

```js
isAutoRefreshEnabled: action.payload.isAutoRefreshEnabled ?? state.isAutoRefreshEnabled,
autoRefreshInternal: action.payload.autoRefreshInterval ?? state.autoRefreshInterval
```

but open to discussion

Reply:
I can go either way. I do like to minimize how much a reducer "assumes" and often make the action explicit so that it instructs specifically what state should be.  Just a thought 🧠